src/devices/MemoryLCD/LSxxxB7DHxx.cs

```diff
@@ -0,0 +1,256 @@
+        #region  Delay constants for LCD timing
+
+        // see datasheet 6-2 Power Supply Sequence
+        private const int T3 = 30; // Release time for initialization of TCOM latch (>30us)
+        private const int T4 = 30; // TCOM polarity initialization time (>30us)
+
+        // see datasheet 6-3 Input Signal Basic Characteristics
+        private const int TsScs = 3; // SCS setup time (>3us)
+        private const int ThScs = 1; // SCS hold time (>1us)
+        #endregion
+
+        #region Screen specification
+
+        /// <summary>
+        /// Memory LCD width pixels
+        /// </summary>
+        public abstract int PixelWidth { get; }
+
+        /// <summary>
+        /// Memory LCD height pixels
+        /// </summary>
+        public abstract int PixelHeight { get; }
+
+        /// <summary>
+        /// Memory LCD bytes per line
+        /// </summary>
+        public int BytesPerLine { get; }
+        #endregion
+
+        #region GPIO config
+
+        /// <summary>
+        /// Chip select signal
+        /// </summary>
+        private readonly int _scs;
+
+        /// <summary>
+        /// Display ON/OFF signal<br/>The display ON/OFF signal is only for display. Data in the memory will be saved at the time of ON/OFF.<br/>When it’s "H", data in the memory will display, when it’s "L", white color will diaplay and data in the memory will be saved.
+        /// </summary>
+        private readonly int _disp;
+
+        /// <summary>
+        /// External COM inversion signal input (H: enable)<br/>When EXTMODE is "Lo", connect the EXTCOMIN to VSS.
+        /// </summary>
+        private readonly int _extcomin;
+        #endregion
+
+        private readonly bool _shouldDispose;
+        private GpioController? _gpio;
+
+        private SpiDevice _spi;
+
+        internal byte[] _lineNumberBuffer;
+        internal byte[] _frameBuffer;
+
+        /// <summary>
+        /// Create a memory LCD device
+        /// </summary>
+        /// <param name="spi">SPI controller<br/><b>ChipSelectLineActiveState</b> must be <b>HIGH</b> or use <paramref name="gpio"/> and <paramref name="chipSelect"/>. See Datasheet 6-3</param>
+        /// <param name="gpio">GPIO controller</param>
+        /// <param name="shouldDispose">True to dispose the Gpio Controller</param>
+        /// <param name="chipSelect">Chip select signal<br/>-1 when using SPI chipSelect line</param>
+        /// <param name="display">Display ON/OFF signal</param>
+        /// <param name="externalCom">External COM inversion signal input</param>
+        internal LSxxxB7DHxx(SpiDevice spi, GpioController? gpio = null, bool shouldDispose = true, int chipSelect = -1, int display = -1, int externalCom = -1)
+        {
+            _spi = spi ?? throw new ArgumentNullException(nameof(spi));
+
+            if (chipSelect != -1 || display != -1 || externalCom != -1)
+            {
+                _shouldDispose = gpio == null || shouldDispose;
+                _gpio = gpio;
```

Comment:
Should we error out in case pins were passed in but gpio is null? or should we in this case just initialize a default gpiocontroller?